### PR TITLE
Improve Jenkins CI

### PR DESCRIPTION
**Pull request by tul1**
I've added a brand new pipeline that is going to use our Docker image and perform more jobs in order to validate the PR's code before enabling the merge operation into master. Also a publishing stage was added, allowing to deploy avalon-rs lib into crate.io when merging from master into stable.

### CODE REVIEW

File: Jenkinsfile
Comment (julianmateu) [inline Jenkinsfile]: Are we merging this if the test are not running? Can't we make this a draft PR until that is solved so we don't loose our existing CI?

File: src/core/election.rs
Comment (julianmateu) [inline src/core/election.rs]: These files shouldn't be in the PR, maybe rebase?

File: Jenkinsfile
Comment (tul1) [reply Jenkinsfile]: Tests were solved inside this pr https://github.com/tul1/avalon-rs/pull/28 . we can wait until it's merge. You cannot make a pr a draft pr.

File: src/core/election.rs
Comment (tul1) [reply src/core/election.rs]: indeed, i'll rebase

File: Jenkinsfile
Comment (julianmateu) [reply Jenkinsfile]: OK. Let's wait until that one is merged

Comment (tul1) [reply Jenkinsfile]: uncommneted

**Review by julianmateu — APPROVED** [review]

**Pull request merged**
